sdks/core/src/main/java/org/gbif/pipelines/core/parsers/location/parser/ContinentParser.java

```diff
@@ -0,0 +1,148 @@
+        }
+      }
+    } else if (lr.getCountryCode() != null) {
+      // No coordinates, but we can check e.g. TURKİYE is either ASIA or EUROPE.
+      Country country = Country.fromIsoCode(lr.getCountryCode());
+
+      if (continent == null) {
+        if (CountryContinentMaps.continentsForCountry(country).size() == 1) {
+          // Set the continent if it's unambiguous, e.g. Switzerland
+          issues.add(CONTINENT_DERIVED_FROM_COUNTRY.name());
+          return ParsedField.success(
+              CountryContinentMaps.continentsForCountry(country).get(0), issues);
+        } else {
+          // Transcontinental country.
+          return parsedContinent;
+        }
+      } else {
+        if (CountryContinentMaps.continentsForCountry(country).contains(continent)) {
+          return parsedContinent;
+        } else {
+          issues.add(CONTINENT_COUNTRY_MISMATCH.name());
+          return ParsedField.success(continent, issues);
+        }
+      }
+    } else {
+      return parsedContinent;
+    }
+  }
+
+  private static Optional<List<Continent>> getContinentFromCoordinates(
+      LatLng latLng, KeyValueStore<LatLng, GeocodeResponse> geocodeKvStore) {
+    if (latLng.isValid()) {
+      GeocodeResponse geocodeResponse = geocodeKvStore.get(latLng);
+      if (geocodeResponse != null && !geocodeResponse.getLocations().isEmpty()) {
+        return Optional.of(
+            geocodeResponse.getLocations().stream()
+                .filter(l -> "Continent".equals(l.getType()))
+                .sorted(Comparator.comparingDouble(Location::getDistance))
+                .map(Location::getId)
+                .map(Continent::fromString)
+                .collect(Collectors.toList()));
+      }
+    }
+    return Optional.empty();
+  }
+
+  private static ParsedField<Continent> parseContinent(
+      ExtendedRecord er, VocabularyParser<Continent> parser, String issue) {
+    Optional<ParseResult<Continent>> parseResultOpt = parser.map(er, parseRes -> parseRes);
+
+    if (!parseResultOpt.isPresent()) {
+      // case when the continent is null in the extended record. We return an issue not to break the
+      // whole interpretation
+      return ParsedField.fail();
+    }
+
+    ParseResult<Continent> parseResult = parseResultOpt.get();
+    ParsedField.ParsedFieldBuilder<Continent> builder = ParsedField.builder();
+    if (parseResult.isSuccessful()) {
+      builder.successful(true);
+      builder.result(parseResult.getPayload());
+    } else {
+      builder.issues(Collections.singleton(issue));
+    }
+    return builder.build();
+  }
+
+  private static Optional<Continent> getContinentResult(
+      ParsedField<Continent> field, Set<String> issues) {
+    if (!field.isSuccessful()) {
+      issues.addAll(field.getIssues());
+    }
+
+    return Optional.ofNullable(field.getResult());
+  }
+
+  public static void accept(Location l) {
+    if (l.getType() != null && l.getDistance() != null && l.getDistance() == 0) {}
```

Comment:
Does this supposed to be empty? Looks strange

Reply:
Oops, copy-paste error. The method isn't used, I'll remove it.